File: dataset_reduzido/avatar/Python/Code/codeforces_340_A.py
def G ( a , b ) :
    while a % b != 0 :
        a , b = b , a % b
    return b
def L ( a , b ) :
    return a * b // G ( a , b )
l = input ( ).split ( ' ' )
x , y , a , b = int ( l [ 0 ] ) , int ( l [ 1 ] ) , int ( l [ 2 ] ) , int ( l [ 3 ] )
lcm = L ( x , y )
print ( b // lcm - ( a - 1 ) // lcm )

File: output/LLaMa/avatar/Python/Java/codeforces_340_A.java
import java.util.Scanner;

public class Main {
    public static int G(int a, int b) {
        while (a % b != 0) {
            int temp = b;
            b = a % b;
            a = temp;
        }
        return b;
    }

    public static int L(int a, int b) {
        return a * b / G(a, b);
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String[] input = scanner.nextLine().split(" ");
        int x = Integer.parseInt(input[0]);
        int y = Integer.parseInt(input[1]);
        int a = Integer.parseInt(input[2]);
        int b = Integer.parseInt(input[3]);
        int lcm = L(x, y);
        System.out.println((b / lcm) - ((a - 1) / lcm));
    }
}